client/src/Pos/Client/Txp/Util.hs

```diff
@@ -199,9 +212,7 @@ makeAbstractTx :: (owner -> TxSigData -> Either e TxInWitness)
                -> Either e TxAux
 makeAbstractTx mkWit txInputs outputs = do
   let
-    tx = UnsafeTx (map snd txInputs) txOutputs txAttributes
-    txOutputs = map toaOut outputs
-    txAttributes = mkAttributes ()
+    Right tx = makeUnsignedAbstractTx txInputs outputs
```

Comment:
This kind of pattern match is really unsafe. Even though we "know" that `makeUnsignedAbstractTx` always returns `Right`, we should encode that in the types instead.